products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp

```diff
@@ -83,16 +75,11 @@ namespace app::bell_settings
                 NumWithStringListItem::NumWithStringSpinner::Range chimeRange;
                 const UTF8 minStr = utils::translate("common_minute_short");
                 chimeRange.push_back(NumWithStringListItem::Value{utils::translate("app_alarm_clock_no_snooze")});
-                for (unsigned int i = 1; i <= val; i++) {
-                    if ((i != Intervals::Interval_1) && (i != Intervals::Interval_2) && (i != Intervals::Interval_5) &&
-                        (i != Intervals::Interval_10) && (i != Intervals::Interval_15) &&
-                        (i != Intervals::Interval_30)) {
-                        continue;
-                    }
-                    chimeRange.push_back(NumWithStringListItem::Value{i, minStr});
-                }
+                std::for_each(allowedIntervals.begin(), allowedIntervals.lower_bound(val), [&](auto val) {
```

Comment:
What happens when, for example: val == 12 (snooze length set in previous window), won't std::lower_bound point to next greater than 12 in the set, which is 15 and add it to the allowable range for chime interval? i think it is not supposed to work this way, for a val of 12 the allowable max. limit for interval should be 10 (always lower than chime length)?

Reply:
No, the lower bound is choosing a lower value between the two closest in the set

Reply:
In that case it's fine, i was looking at https://www.geeksforgeeks.org/set-lower_bound-function-in-c-stl/ - maybe there is an error there

Reply:
can we resolve this conversation?